server/boot/user.js

```diff
@@ -56,6 +55,61 @@ function encodeFcc(value = '') {
   return replaceScriptTags(replaceFormAction(value));
 }
 
+function isAlgorithm(challenge) {
+  // test if name starts with hike/waypoint/basejump/zipline
+  // fix for bug that saved different challenges with incorrect
+  // challenge types
+  return !(/^(waypoint|hike|zipline|basejump)/i).test(challenge.name) &&
```

Comment:
But we also have to take into account that there are some challenges that didn't get right challengeType. See isAlgorithm function.